int32 machine_id
float32 pos_x
float32 pos_y
float32 radius
bool is_working
string[4] wanted_resource
string[4] product


